analysis/monkmistweaver/src/modules/core/HotTrackerMW.tsx

```diff
@@ -23,37 +23,39 @@ class HotTrackerMW extends HotTracker {
     this.upwellingActive = selectedCombatant.hasTalent(SPELLS.UPWELLING_TALENT.id);
   }
 
-  calculateMaxDuration(event: ApplyBuffEvent & { selectedCombatant: Combatant }) {
+  calculateMaxDuration(event: ApplyBuffEvent, combatant: Combatant) {
     const spellId = event.ability.guid;
     if (spellId === SPELLS.ENVELOPING_MIST.id) {
-      return (ENV_BASE_DURATION + (this.mistwrapActive ? MISTWRAP : 0)) * 2;
+      return (
+        (ENV_BASE_DURATION + (combatant.hasTalent(SPELLS.MIST_WRAP_TALENT.id) ? MISTWRAP : 0)) * 2
+      );
     }
     if (spellId === SPELLS.ESSENCE_FONT_BUFF.id) {
-      return (EF_BASE_DURATION + (this.upwellingActive ? UPWELLING : 0)) * 2;
```

Comment:
A very strange issue in Typescript was making `this` undefined inside this function. I couldn't figure how to resolve it so I simply changed references to `this` to get at talents through the passed in Combatant instead